apps/studio/components/to-be-cleaned/Storage/StorageSettings/S3Connection.tsx

```diff
@@ -0,0 +1,337 @@
+import { useParams } from 'common'
+import { useProjectContext } from 'components/layouts/ProjectLayout/ProjectContext'
+import { FormHeader } from 'components/ui/Forms'
+import { useStorageCredentialsCreateMutation } from 'data/storage/storage-credentials-create-mutation'
+import { useStorageCredentialsQuery } from 'data/storage/storage-credentials-query'
+import React from 'react'
+import toast from 'react-hot-toast'
+import {
+  Button,
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogTitle,
+  DialogTrigger,
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+  IconMoreVertical,
+  IconTrash,
+  Input,
+  Label_Shadcn_,
+  cn,
+} from 'ui'
+import Table from 'components/to-be-cleaned/Table'
+import Panel from 'components/ui/Panel'
+import { useStorageCredentialsDeleteMutation } from 'data/storage/storage-credentials-delete-mutation'
+import { GenericSkeletonLoader } from 'ui-patterns'
+import CopyButton from 'components/ui/CopyButton'
+
+type Props = {}
+
+export const S3Connection = (props: Props) => {
+  const [openCreateCred, setOpenCreateCred] = React.useState(false)
+  const [showSuccess, setShowSuccess] = React.useState(false)
+  const [openDeleteDialog, setOpenDeleteDialog] = React.useState(false)
+  const [deleteCredId, setDeleteCredId] = React.useState<string | null>(null)
+
+  const { ref: projectRef } = useParams()
+  const { project, isLoading: projectIsLoading } = useProjectContext()
+
+  const { data: storageCreds, ...storageCredsQuery } = useStorageCredentialsQuery({
+    projectRef,
+  })
+
+  const hasStorageCreds = storageCreds?.data && storageCreds.data.length > 0
+
+  const createStorageCreds = useStorageCredentialsCreateMutation({
+    projectRef,
+  })
+
+  const deleteStorageCreds = useStorageCredentialsDeleteMutation({
+    projectRef,
+  })
+
+  function getConnectionURL() {
+    const baseUrl = process.env.NEXT_PUBLIC_SITE_URL
```

Comment:
This won't work in production, I think. In production this is set to `supabase.com` but the instance url ends in `supabase.co`, right? You need to use `useProjectApiQuery` (check `useRealtimeMessages.tsx` how it's used).